### ISSUE TITLE
rails 7.2: can't chain `reload!` in console anymore?

### ISSUE THREAD

**Issue by josh-m-sharpe** [rails/rails#53404]
In ~all~ many many versions of rails up to and including 7.1.4.1 you could, in rails console, set up a one liner beginning with reload! for example: `reload! ; u = User.first ; u.thing.im.hacking.on` and it worked.  You could then up/enter/up/enter/dev/up/enter/repeat/up/enter/develop/break/things/up/enter/etc to your heart's content.

It would seem that `reload!` no longer can be chained with `;` in rails console. Now, `reload! ; anything_else` seems to skip everything after the reload:

rails 7.2.1.1:
```
3.3.5 :001 > reload! ; u = User.first
Reloading...
3.3.5 :002 >
```

rails 7.1.4.1:
```
3.3.5 :001 > reload! ; u = User.first
Reloading...
  User Load (0.3ms)  SELECT "users".* FROM "users" ORDER BY "users"."id" ASC LIMIT $1  [["LIMIT", 1]]
 =>
#<User:0x000000013d9304e8
...
3.3.5 :002 >
```

(Also `reload!` is bolded in the above (7.2.1.1) output.  So, it must have gotten some special treatment recently.)

This was an extremely nice 'feature' if undocumented.  It allows for quickly repeating the same series of commands with developing or debugging.  The new behavior makes this more difficult.

This appears to have been introduced sometimes after > 7.1.4.1 and <= 7.2.1.1

### Expected behavior
`reload!` acted like a normal method I could chain with `;` in console the way it used to.

### Actual behavior
It no longer allows for methods to be called after it.

### System configuration
**Rails version**:
7.2.1.1
**Ruby version**:
3.3.5

Confirmed this with fresh 7.2 rails app on ruby 3.3.5.  So you can repro this with... `rails new` and `rails console`

**Comment by josh-m-sharpe** [comment]
Maybe, possibly this? https://github.com/rails/rails/commit/484c026f9818aff03b41de3ff35d4cfff8c45bf0 @st0012

**Comment by jeromedalbert** [comment]
Thanks for reporting the issue. This is a duplicate of https://github.com/rails/rails/issues/52982, and was fixed in https://github.com/rails/rails/pull/52996, so it works in the Rails 8 prerelease.

Wondering if it could be back ported to Rails 7.2 though

**Comment by skipkayhil** [comment]
It was backported in https://github.com/rails/rails/commit/6662c51d6ee899709f1b287046ccedf75e66b89a when #52996 was merged and will be released in 7.2.2 (or you can use `7-2-stable`)
